clients/pkg/promtail/positions/write_positions.go

```diff
@@ -0,0 +1,27 @@
+// +build !windows
+package positions
+import (
+	"os"
+	"path/filepath"
+
+	renameio "github.com/google/renameio/v2"
+	yaml "gopkg.in/yaml.v2"
+)
+
+func writePositionFile(filename string, positions map[string]string) error {
+	buf, err := yaml.Marshal(File{
+		Positions: positions,
+	})
+	if err != nil {
+		return err
+	}
+
+	target := filepath.Clean(filename)
+
+	err = renameio.WriteFile(target, buf, os.FileMode(positionFileMode))
+	if err != nil {
+		return err
+	}
+
+	return nil
```

Comment:
You could simply return the return value from `renameio.WriteFile()`

```suggestion
	return renameio.WriteFile(target, buf, os.FileMode(positionFileMode))
```

Reply:
:see_no_evil: